src/disk_interface.cc

```diff
@@ -100,19 +100,20 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
       static_cast<FINDEX_INFO_LEVELS>(1);
   FINDEX_INFO_LEVELS level =
       can_use_basic_info ? kFindExInfoBasic : FindExInfoStandard;
-  WIN32_FIND_DATAA ffd;
-  HANDLE find_handle = FindFirstFileExA((dir + "\\*").c_str(), level, &ffd,
-                                        FindExSearchNameMatch, NULL, 0);
+
+  WIN32_FIND_DATA ffd;
+  HANDLE find_handle = FindFirstFileEx(Utf8ToWide(dir + "\\*").c_str(), level,
+                                       &ffd, FindExSearchNameMatch, NULL, 0);
 
   if (find_handle == INVALID_HANDLE_VALUE) {
     DWORD win_err = GetLastError();
     if (win_err == ERROR_FILE_NOT_FOUND || win_err == ERROR_PATH_NOT_FOUND)
```

Comment:
Are these really the only possible failures?

Reply:
would anything from this PR change that though?

Reply:
That's not obvious one way or another.

Among other things, a different function is being called.

But regardless of whether the change to wstring would change the error outcomes, this PR is making substantial changes to the code surrounding this error check, and so should also make sure the error check remains, or becomes, valid.

Reply:
Looking at the documentation reveals no obvious difference in the return value, so the check should still be valid.